File: app/src/main/java/buggy/plandroid/com/plandroidbugs/MainActivity.java
package buggy.plandroid.com.plandroidbugs;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;

import buggy.plandroid.com.plandroidbugs.app.PlanDroidApp;
import buggy.plandroid.com.plandroidbugs.room.UserDao;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;


public class MainActivity extends AppCompatActivity {

    @Override protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        RecyclerView recyclerView = findViewById(R.id.recycler);
        UserAdapter userAdapter = new UserAdapter();
        recyclerView.setAdapter(userAdapter);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        UserDao userDao = PlanDroidApp.getApp(getApplicationContext()).getDatabase().userDao();

        userDao.getAll().observeOn(AndroidSchedulers.mainThread()).subscribe(it -> userAdapter.setUsers(it));
    }
}

File: app/src/main/java/buggy/plandroid/com/plandroidbugs/MainActivity.kt
package buggy.plandroid.com.plandroidbugs

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import buggy.plandroid.com.plandroidbugs.app.PlanDroidApp.Companion.getApp
import io.reactivex.android.schedulers.AndroidSchedulers

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val userAdapter = UserAdapter()
        findViewById<RecyclerView>(R.id.recycler).apply {
            adapter = userAdapter
            layoutManager = LinearLayoutManager(this@MainActivity)
        }
        val userDao = getApp(applicationContext).database.userDao()

        // Listens to changes to the User database table.
        userDao.all().observeOn(AndroidSchedulers.mainThread())
            .subscribe { userAdapter.setUsers(it) }

    }
}